File: sunnad-ios/sunnad-ios/Domain/Repositories/HabitsRepository.swift
import Foundation

protocol HabitsRepository: Sendable {
    func fetchHabits(includeArchived: Bool) async throws -> [Habit]
    func fetchDueHabits(on day: Date, calendar: Calendar, timeZone: TimeZone) async throws -> [Habit]
    func saveHabit(_ habit: Habit) async throws
    func deleteHabit(id: UUID) async throws
}

File: sunnad-android/app/src/main/java/com/arystan/almasuly/sunnadandroid/domain/repository/HabitsRepository.kt
package com.arystan.almasuly.sunnadandroid.domain.repository

import com.arystan.almasuly.sunnadandroid.core.model.Habit
import java.time.LocalDate
import java.util.UUID

interface HabitsRepository {
    suspend fun fetchHabits(includeArchived: Boolean): List<Habit>
    suspend fun fetchDueHabits(day: LocalDate): List<Habit>
    suspend fun saveHabit(habit: Habit)
    suspend fun deleteHabit(id: UUID)
}
